src/signatureInfoView.ts

```diff
@@ -0,0 +1,312 @@
+import * as vscode from 'vscode';
+import { Renderer } from './renderer';
+
+enum UpdateMode {
+	Live = 'live',
+	Sticky = 'sticky',
+}
+
+export class SignatureInfoViewViewProvider implements vscode.WebviewViewProvider {
+
+	public static readonly viewType = 'docsView.signatureinfo';
+
+	private static readonly pinnedContext = 'docsView.signatureInfoView.isPinned';
+
+	private readonly _disposables: vscode.Disposable[] = [];
+
+	private readonly _renderer = new Renderer();
+
+	private _view?: vscode.WebviewView;
+	private _currentCacheKey: CacheKey = cacheKeyNone;
+	private _loading?: { cts: vscode.CancellationTokenSource }
+
+	private _updateMode = UpdateMode.Live;
+	private _pinned = false;
+
+	constructor(
+		private readonly _extensionUri: vscode.Uri,
+	) {
+		vscode.window.onDidChangeActiveTextEditor(() => {
+			this.update();
+		}, null, this._disposables);
+
+		vscode.window.onDidChangeTextEditorSelection(() => {
+			this.update();
+		}, null, this._disposables);
+
+		this._renderer.needsRender(() => {
+			this.update(/* force */ true);
+		}, undefined, this._disposables);
+
+		vscode.workspace.onDidChangeConfiguration(() => {
+			this.updateConfiguration();
+		}, null, this._disposables);
+
+		this.updateConfiguration();
+		this.update();
+	}
+
+	dispose() {
+		let item: vscode.Disposable | undefined;
+		while ((item = this._disposables.pop())) {
+			item.dispose();
+		}
+	}
+
+	public resolveWebviewView(
+		webviewView: vscode.WebviewView,
+		_context: vscode.WebviewViewResolveContext,
+		_token: vscode.CancellationToken,
+	) {
+		this._view = webviewView;
+
+		webviewView.webview.options = {
+			enableScripts: true,
+			localResourceRoots: [
+				vscode.Uri.joinPath(this._extensionUri, 'media')
+			]
+		};
+
+		webviewView.onDidChangeVisibility(() => {
+			if (this._view?.visible) {
+				this.update(/* force */ true);
+			}
+		});
+
+		webviewView.onDidDispose(() => {
+			this._view = undefined;
+		});
+
+		this.updateTitle();
+		webviewView.webview.html = this._getHtmlForWebview(webviewView.webview);
+
+		this.update(/* force */ true);
+	}
+
+	public pin() {
```

Comment:
Looks like pinning isn't hooked up at the moment so it ether needs to be added or this code should be removed

Reply:
Any hints on how to use CSS?

Reply:
Will hook it up.

Reply:
This is done.